src/backend/marsha/core/serializers/video.py

```diff
@@ -434,6 +434,128 @@ def validate(self, attrs):
+            "active_stamp",
+            "filename",
+            "id",
+            "is_ready_to_show",
+            "nb_pages",
+            "upload_state",
+            "urls",
+            "video",
+        )
+
+    active_stamp = TimestampField(
+        source="uploaded_on", required=False, allow_null=True, read_only=True
+    )
+    filename = serializers.SerializerMethodField()
+    is_ready_to_show = serializers.BooleanField(read_only=True)
+    urls = serializers.SerializerMethodField()
+
+    def create(self, validated_data):
+        """Force the video field to the video of the JWT Token if any.
+
+        Parameters
+        ----------
+        validated_data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        dictionary
+            The "validated_data" dictionary is returned after modification.
+
+        """
+        # user here is a video as it comes from the JWT
+        # It is named "user" by convention in the `rest_framework_simplejwt` dependency we use.
+        user = self.context["request"].user
+        # Set the video field from the payload if there is one and the user is identified
+        # as a proper user object through access rights
+        if (
+            self.initial_data.get("video")
+            and user.token.get("user")
+            and user.token["resource_id"] == user.token.get("user", {}).get("id")
+        ):
+            validated_data["video_id"] = self.initial_data.get("video")
+        # If the user just has a token for a video, force the video ID on the shared live media
+        if not validated_data.get("video_id") and isinstance(user, TokenUser):
+            validated_data["video_id"] = user.id
+        return super().create(validated_data)
+
+    def get_filename(self, obj):
+        """Filename of the shared live media."""
+        return (
+            self._get_filename(obj.title, extension=obj.extension)
+            if obj.uploaded_on
+            else None
+        )
+
+    def get_urls(self, obj):
+        """urls for each media's pages."""
+
+        if obj.uploaded_on is None or obj.nb_pages is None:
+            return None
+
+        urls = {}
+
+        stamp = time_utils.to_timestamp(obj.uploaded_on)
+        base = (
+            f"{settings.AWS_S3_URL_PROTOCOL}://{settings.CLOUDFRONT_DOMAIN}/"
+            f"{obj.video.pk}/sharedlivemedias/{obj.pk}/{stamp}"
+        )
+
+        cloudfront_signer = None
+        if settings.CLOUDFRONT_SIGNED_URLS_ACTIVE:
+            date_less_than = timezone.now() + timedelta(
+                seconds=settings.CLOUDFRONT_SIGNED_URLS_VALIDITY
+            )
+            cloudfront_signer = CloudFrontSigner(
+                settings.CLOUDFRONT_ACCESS_KEY_ID, cloudfront_utils.rsa_signer
+            )
+
+        pages = {}
+        for i in range(obj.nb_pages):
+            page_number = i + 1
+            url = f"{base:s}/{page_number:d}.png"
```

Comment:
Shouldn't the extension be `svg` ? Or should  we make the output conversion format configurable ?

Reply:
Probably, I will change it once the `lambda` part of this feature done.